lms/djangoapps/verified_track_content/tasks.py

```
@@ -26,28 +26,35 @@ def sync_cohort_with_mode(course_id, user_id, verified_cohort_name, default_coho
-            user.username, verified_cohort.name, course_id
-        )
-        add_user_to_cohort(verified_cohort, user.username)
-    elif enrollment.mode != CourseMode.VERIFIED and current_cohort.id == verified_cohort.id:
-        default_cohort = get_cohort_by_name(course_key, default_cohort_name)
-        LOGGER.info(
-            "MOVING_TO_DEFAULT: Moving user '%s' to the default cohort '%s' for course '%s'",
-            user.username, default_cohort.name, course_id
-        )
-        add_user_to_cohort(default_cohort, user.username)
-    else:
-        LOGGER.info(
-            "NO_ACTION_NECESSARY: No action necessary for user '%s' in course '%s' and enrollment mode '%s'. "
-            "The user is already in cohort '%s'.",
-            user.username, course_id, enrollment.mode, current_cohort.name
+        if enrollment.mode == CourseMode.VERIFIED and (current_cohort.id != verified_cohort.id):
+            LOGGER.info(
+                "MOVING_TO_VERIFIED: Moving user '%s' to the verified cohort '%s' for course '%s'",
+                user.username, verified_cohort.name, course_id
+            )
+            add_user_to_cohort(verified_cohort, user.username)
+        elif enrollment.mode != CourseMode.VERIFIED and current_cohort.id == verified_cohort.id:
+            default_cohort = get_cohort_by_name(course_key, default_cohort_name)
+            LOGGER.info(
+                "MOVING_TO_DEFAULT: Moving user '%s' to the default cohort '%s' for course '%s'",
+                user.username, default_cohort.name, course_id
+            )
+            add_user_to_cohort(default_cohort, user.username)
+        else:
+            LOGGER.info(
+                "NO_ACTION_NECESSARY: No action necessary for user '%s' in course '%s' and enrollment mode '%s'. "
+                "The user is already in cohort '%s'.",
+                user.username, course_id, enrollment.mode, current_cohort.name
+            )
+    except Exception as exc:
```

Might not apply as much to this specific use case - but would a more specific exception type be better here? Bulk email, as an example, will catch-and-retry for some errors, and just fail for all others. Is that sort of handling useful here, or is it simple enough that a bare except with 4 retries is fine?